src/crypto/hsm/nxp/README.md

```diff
@@ -0,0 +1,75 @@
+# Using SE05X for crypto operations
+
+## Introduction
+
+CHIPCryptoPALHsm_SE05X.cpp file provides the integration of secure element (SE05x) in crypto layer of CHIP stack.
+By enabling the required directives in CHIPCryptoPALHsm_config.h header file,  required crypto operations can be performed using SE05x.
+By default, the secure element is enabled only for PAKE (Verifier role) operations.
+
+Following table shows the operations supported by secure element.
+
++-------------------------+-----------+----------------+---------------------------------+------------+----------------------------+
+| Operation               | Software  | Default config | Note                            | Re-entrant | Directive to enabled in    |
```

Comment:
I'm glad reentrancy is documented.  However, can we not add locking to the new critical paths to maintain thread safety for `Spake2p` the session objects that use this?

Reply:
Thanks. Let me have a closer look on how best to handle this.

Reply:
I would think the most straight forward thing to do would be to add a static mutex to the `Spake2p` definition.  Then calls to the HSM can either block on the mutex, or a try-lock approach could used to fall back to software implementations.

The former approach enforces use of the HSM always.  The latter approach gives an opportunity for concurrency.

Reply:
We now have taken care of Re-entrancy for the Spake2+ HSM based implementation. We moved a static data structure into the Spake Class. A Mutex is now used to protect a shared object

Reply:
Thank you!